src/main/java/frc/robot/subsystems/shooter/ShooterIOTalonFX.java

```diff
@@ -0,0 +1,294 @@
+package frc.robot.subsystems.shooter;
+
+import static frc.robot.subsystems.shooter.ShooterConstants.*;
+
+import com.ctre.phoenix.sensors.CANCoderConfiguration;
+import com.ctre.phoenix6.BaseStatusSignal;
+import com.ctre.phoenix6.StatusSignal;
+import com.ctre.phoenix6.configs.CANcoderConfiguration;
+import com.ctre.phoenix6.configs.CurrentLimitsConfigs;
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.controls.VelocityTorqueCurrentFOC;
+import com.ctre.phoenix6.hardware.CANcoder;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.InvertedValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+
+import edu.wpi.first.wpilibj.DigitalInput;
+import frc.lib.team3061.RobotConfig;
+import frc.lib.team6328.util.TunableNumber;
+
+public class ShooterIOTalonFX implements ShooterIO {
+
+  // Using VelocityTorqueCurrentFOC to set the velocity of the motors
+  private VelocityTorqueCurrentFOC shootMotorLeftVelocityRequest;
+  private VelocityTorqueCurrentFOC shootMotorRightVelocityRequest;
+  private VelocityTorqueCurrentFOC drumMotorVelocityRequest;
+  private VelocityTorqueCurrentFOC dunkerMotorVelocityRequest;
+  // Using motion magic for the Angle Motor
+
+  // Using StatusSignal to get the stator current of the motors
+  private StatusSignal<Double> shootMotorLeftStatorCurrentStatusSignal;
+  private StatusSignal<Double> shootMotorRightStatorCurrentStatusSignal;
+  private StatusSignal<Double> drumMotorStatorCurrentStatusSignal;
+  private StatusSignal<Double> dunkerMotorStatorCurrentStatusSignal;
+  private StatusSignal<Double> angleMotorStatorCurrentStatusSignal;
+
+  // Using StatusSignal to get the supply current of the motors
+  private StatusSignal<Double> shootMotorLeftSupplyCurrentStatusSignal;
+  private StatusSignal<Double> shootMotorRightSupplyCurrentStatusSignal;
+  private StatusSignal<Double> drumMotorSupplyCurrentStatusSignal;
+  private StatusSignal<Double> dunkerMotorSupplyCurrentStatusSignal;
+  private StatusSignal<Double> angleMotorSupplyCurrentStatusSignal;
+
+  // Using StatusSignal to get the velocity of the motors
+  private StatusSignal<Double> shootMotorLeftVelocityStatusSignal;
+  private StatusSignal<Double> shootMotorRightVelocityStatusSignal;
+  private StatusSignal<Double> drumMotorVelocityStatusSignal;
+  private StatusSignal<Double> dunkerMotorVelocityStatusSignal;
+
+  // Creates the reference velocity request for the motors
+  private double shootMotorLeftRequestedVelocityRPS;
```

Comment:
We can get these using reference status signals; so, the instance variables aren't needed. You could add status signals for the velocity references.

However, we will control the position, not velocity of the angle motor.